pypesto/profile/profile_next_guess.py

```diff
@@ -429,14 +525,36 @@ def do_line_search(
         The parameter estimation problem.
     options:
         Profile likelihood options.
+    min_step_increase_factor:
+        Factor to increase the minimal step size bound.
+    max_step_reduce_factor:
+        Factor to reduce the maximal step size bound.
 
     Returns
     -------
     Parameter vector that is expected to yield the objective function value
     closest to `next_obj_target`.
     """
-    # Was the initial step too big or too small?
-    direction = "decrease" if next_obj_target < next_obj else "increase"
+    decreasing_to_low_target = False
+    decreasing_to_high_target = False
+
+    # Determine the direction of the step
+    if next_obj >= current_obj:
+        next_obj_target = high_next_obj_target
+        if next_obj >= high_next_obj_target:
+            direction = "decrease"
+            decreasing_to_high_target = True
+        else:
+            direction = "increase"
+
+    elif next_obj < current_obj:
+        next_obj_target = low_next_obj_target
+        if next_obj <= low_next_obj_target:
+            direction = "decrease"
+            decreasing_to_low_target = True
+        else:
+            direction = "increase"
```

Comment:
If the initial guess is between the lower and higher target, we increase step size till we hit any of them. (exit conditions below)

If the initial guess is below lower target or above higher target, we decrease the step size till we hit the lower or higher target, respectively.